internal/infrastructure/storage/db/badger/trade_repository_impl.go

```diff
@@ -210,34 +210,45 @@ func (t tradeRepositoryImpl) getTrade(
 	ctx context.Context,
 	ID uuid.UUID,
 ) (*domain.Trade, error) {
-	var trade domain.Trade
+	var trades []domain.Trade
 	var err error
+	query := badgerhold.Where("ID").Eq(ID)
 	if ctx.Value("tx") != nil {
 		tx := ctx.Value("tx").(*badger.Txn)
-		err = t.store.TxGet(tx, ID, &trade)
+		err = t.store.TxFind(tx, &trades, query)
```

Comment:
whats difference between Find & Get?

I recall some badger bug or something

Reply:
With Find we use a sql-like query to retrieve an element, while Get returns the value that matches the given key. I didn't go in deep but it's likely that Find returns a value that matches the given condition without checking the key like Get does instead.

The real bug here is using a uuid.UUID as key instead of a built-in type like string. This update prevents very weird behaviors by the repository btw. (Un?)Luckily for us, no providers have ever suffered from this.

Reply:
nACK
Get is much faster, since it is searching by key, i got bellow benchmarks:
WithFind            1468791346 ns/op
WithGet		  66934           ns/op

Reply:
may or may not be related https://github.com/tdex-network/tdex-daemon/pull/454

Reply:
Same here: this will be fixed, ie. restored to use Get, in next PR, but for now it's need to prevent unexpected behaviors.

Reply:
add a comment?

Reply:
Done. Sorry i completely miss this 🙏